contracts/mock-hook-consumer/src/contract.rs

```diff
@@ -0,0 +1,183 @@
+                ))?,
+                funds: vec![],
+            }),
+            CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: task_queue.to_string(),
+                msg: to_json_binary(&lavs_apis::tasks::ExecuteMsg::Custom(
+                    lavs_apis::tasks::CustomExecuteMsg::AddHook(TaskHookType::Timeout),
+                ))?,
+                funds: vec![],
+            }),
+        ];
+
+        Ok(Response::default().add_messages(msgs))
+    }
+
+    /// For a task created, we want to increase our created counter.
+    pub fn task_created(
+        deps: DepsMut,
+        _env: Env,
+        _info: MessageInfo,
+        _task: TaskResponse,
+    ) -> StdResult<Response> {
+        CREATED_COUNT.update(deps.storage, |x| -> StdResult<_> { Ok(x + Uint128::one()) })?;
+
+        Ok(Response::default())
+    }
+
+    /// For the task completed, we want to create another task on the task queue.
+    pub fn task_completed(
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        task: TaskResponse,
+    ) -> StdResult<Response> {
+        let task_queue = info.sender;
+
+        // Attempt to deserialize the task response
+        let response: TaskResponseData =
+            serde_json::from_value(task.result.expect("Result is not available")).map_err(|e| {
+                StdError::generic_err(format!(
+                    "Could not deserialize input request from JSON: {}",
+                    e
+                ))
+            })?;
+
+        // Query requestor config
+        let config: ConfigResponse = deps.querier.query_wasm_smart(
+            task_queue.to_string(),
+            &lavs_apis::tasks::QueryMsg::Custom(lavs_apis::tasks::CustomQueryMsg::Config {}),
+        )?;
+
+        // Get amount to send
+        let funds = match config.requestor {
+            Requestor::Fixed(addr) => {
+                if addr == env.contract.address.to_string() {
+                    Ok(vec![])
+                } else {
+                    Err(StdError::generic_err(
+                        "Contract is not authorized to create tasks",
+                    ))
+                }
+            }
+            Requestor::OpenPayment(coin) => Ok(vec![coin]),
+        }?;
+
+        // Construct a new request to square the square result
+        let request = TaskRequestData { x: response.y };
+
+        let msg = CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: task_queue.to_string(),
+            msg: to_json_binary(&lavs_apis::tasks::ExecuteMsg::Custom(
+                lavs_apis::tasks::CustomExecuteMsg::Create {
+                    description: task.description,
+                    timeout: None,
+                    payload: serde_json::to_value(request).map_err(|e| {
+                        StdError::generic_err(format!(
+                            "Could not serialize request into JSON: {}",
+                            e
+                        ))
```

Comment:
This is one thing that puzzles me, I assume that this is just an example of an operation handled automatically after the one from the queue is finished?

Reply:
Yeah, this was an example of a task completion triggering the creation of another task: https://github.com/Lay3rLabs/avs-toolkit/issues/76#issuecomment-2397918372

Reply:
I see, thank you :+1: